### PR TITLE
test: add command line indicator option checking to fipsinstall

### PR DESCRIPTION
**Pull request by paulidale**
Validate that the relevant options are on when -pedantic is specified, off when it isn't and can be given to enable the setting.

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/recipes/03-test_fipsinstall.t [hunk 1]
```diff
@@ -35,7 +35,35 @@ my @pedantic_fail =
...
+        ( 'signature_digest_check',         'signature-digest-check' ),
+        ( 'hkdf_digest_check',              'hkdf-digest-check' ),
+        ( 'tls13_kdf_digest_check',         'tls13-kdf-digest-check' ),
+        ( 'tls1_prf_digest_check',          'tls1-prf-digest-check' ),
+        ( 'sshkdf_digest_check',            'sshkdf-digest-check' ),
+        ( 'sskdf_digest_check',             'sskdf-digest-check' ),
+        ( 'x963kdf_digest_check',           'x963kdf-digest-check' ),
+        ( 'dsa_sign_disabled',              'dsa-sign-disabled' ),
+        ( 'tdes_encrypt_disabled',          'tdes-encrypt-disabled' ),
+        ( 'rsa_pkcs15_padding_disabled',    'rsa-pkcs15-padding-disabled' ),
```
Comment (slontis) [inline test/recipes/03-test_fipsinstall.t:53]: padding will become pad due to another PR..

**Comment by paulidale** [comment]
Ping @openssl/otc @openssl/committers for reviews please.

**Review by nhorman — APPROVED** [review]
Can you link to the subsequent PR that changes padding to pad so we know to block it until this is merged please?

**Comment by paulidale** [comment]
It's #25162 which is still in draft.

**Review by slontis — APPROVED** [review]

**Comment by paulidale** [comment]
Kicking tests before merge.

**Pull request closed without merging**